command_line/sweep_to_stills.py

```diff
@@ -0,0 +1,230 @@
+                # is rotated by a further half oscillation step.
+                A = (
+                    goniometer_axis.axis_and_angle_as_r3_rotation_matrix(
+                        angle=experiment.scan.get_angle_from_array_index(i_scanpoint)
+                        + (step / 2),
+                        deg=True,
+                    )
+                    * goniometer_setting_matrix
+                    * sqr(experiment.crystal.get_A_at_scan_point(i_scanpoint))
+                )
+                ori = crystal_orientation.crystal_orientation(
+                    A, crystal_orientation.basis_type.reciprocal
+                )
+                direct_matrix = ori.direct_matrix()
+                real_a = direct_matrix[0:3]
+                real_b = direct_matrix[3:6]
+                real_c = direct_matrix[6:9]
+                crystal = MosaicCrystalSauter2014(
+                    real_a, real_b, real_c, experiment.crystal.get_space_group()
+                )
+
+                # Copy in mosaic parameters if available
+                if params.output.domain_size_ang is None and hasattr(
+                    experiment.crystal, "get_domain_size_ang"
+                ):
+                    crystal.set_domain_size_ang(
+                        experiment.crystal.get_domain_size_ang()
+                    )
+                elif params.output.domain_size_ang is not None:
+                    crystal.set_domain_size_ang(params.output.domain_size_ang)
+
+                if params.output.half_mosaicity_deg is None and hasattr(
+                    experiment.crystal, "get_half_mosaicity_deg"
+                ):
+                    crystal.set_half_mosaicity_deg(
+                        experiment.crystal.get_half_mosaicity_deg()
+                    )
+                elif params.output.half_mosaicity_deg is not None:
+                    crystal.set_half_mosaicity_deg(params.output.half_mosaicity_deg)
+
+                new_experiment = Experiment()
+                new_experiment.detector = experiment.detector
+                new_experiment.beam = experiment.beam
+                new_experiment.crystal = crystal
+                new_experiment.scan = None
+                new_experiment.goniometer = None
+                new_experiment.imageset = experiment.imageset.as_imageset()[
+                    i_scanpoint : i_scanpoint + 1
+                ]
+                new_experiments.append(new_experiment)
+
+                # Each reflection in a 3D shoebox can be found on multiple images. Slice the reflections
+                # such that any reflection on this scan point is included with this image
+                new_id = len(new_experiments) - 1
+                subrefls = refls.select((i_scanpoint >= z1) & (i_scanpoint < z2))
+                for refl in subrefls:  # this will break when #908 is merged
```

Comment:
All the more reason for requiring a test when the pull request contains comments like this!
Would it be possible to re-write this loop to avoid breaking when #908 is merged?